src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -344,6 +360,34 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
 
                     if (isCurrentPrimaryStale)
                     {
+                        if (_maxElectionInfo == null)
+                        {
+                            _sdamInformationEventHandler?.Invoke(new SdamInformationEvent(() =>
+                                $"Initializing (maxSetVersion, maxElectionId) to " + 
+                                $"({args.NewServerDescription.ReplicaSetConfig.Version}, " +
+                                $"{args.NewServerDescription.ElectionId}) from " +
+                                $"replica set primary {TraceSourceEventHelper.Format(newServerDescription.EndPoint)}"));
+                        }
+                        else
+                        {
+                            if (_maxElectionInfo.IsSetVersionStale(args.NewServerDescription.ReplicaSetConfig.Version.Value))
+                            {
+                                _sdamInformationEventHandler?.Invoke(new SdamInformationEvent(() =>
```

Comment:
Equivalent lines in Java: https://github.com/mongodb/mongo-java-driver/blob/3e777058b556d67c68b084372f940f80eb0e9398/driver-core/src/main/com/mongodb/internal/connection/MultiServerCluster.java#L262-L263